objects/src/accounts/component/template/storage_entry/mod.rs

```diff
@@ -0,0 +1,715 @@
+    ) -> Result<Self, AccountComponentTemplateError> {
+        if slots.len() != values.len() {
+            return Err(AccountComponentTemplateError::MultiSlotArityMismatch);
+        }
+
+        for window in slots.windows(2) {
+            if window[1] != window[0] + 1 {
+                return Err(AccountComponentTemplateError::NonContiguousSlots(
+                    window[0], window[1],
+                ));
+            }
+        }
+
+        Ok(StorageEntry::MultiSlot {
+            name: name.into(),
+            description: description.map(Into::<String>::into),
+            slots,
+            values: values.into_iter().map(Into::into).collect(),
+        })
+    }
+
+    /// Returns the slot indices that the storage entry covers.
+    pub fn slot_indices(&self) -> &[u8] {
+        match self {
+            StorageEntry::MultiSlot { slots, .. } => slots.as_slice(),
+            StorageEntry::Value { slot, .. } => core::slice::from_ref(slot),
+            StorageEntry::Map { slot, .. } => core::slice::from_ref(slot),
+        }
+    }
+
+    /// Returns the name of the storage entry.
+    pub fn name(&self) -> &str {
+        match self {
+            StorageEntry::Value { name, .. } => name.as_str(),
+            StorageEntry::Map { name, .. } => name.as_str(),
+            StorageEntry::MultiSlot { name, .. } => name.as_str(),
+        }
+    }
+
+    /// Returns the optional description of the storage entry.
+    pub fn description(&self) -> Option<&str> {
+        match self {
+            StorageEntry::Value { description, .. } => description.as_deref(),
+            StorageEntry::Map { description, .. } => description.as_deref(),
+            StorageEntry::MultiSlot { description, .. } => description.as_deref(),
+        }
+    }
+
+    /// Returns all the `WordRepresentation` values covered by this entry.
+    /// For `Value` entries, this returns a single-element slice.
+    /// For `MultiSlot` entries, this returns all values.
+    /// For `Map` entries, since they're key-value pairs, return an empty slice.
+    pub fn word_values(&self) -> &[WordRepresentation] {
+        match self {
+            StorageEntry::Value { value, .. } => core::slice::from_ref(value),
+            StorageEntry::MultiSlot { values, .. } => values.as_slice(),
+            StorageEntry::Map { .. } => &[],
+        }
+    }
+
+    /// Returns the map entries for a `Map` variant as a slice.
+    /// Returns an empty slice for non-map variants.
+    pub fn map_entries(&self) -> &[MapEntry] {
+        match self {
+            StorageEntry::Map { values, .. } => values.as_slice(),
+            StorageEntry::Value { .. } => &[],
+            StorageEntry::MultiSlot { .. } => &[],
+        }
+    }
+
+    /// Returns an iterator over all of the storage entries's template keys.
+    // TODO: Should template keys be typed?
+    pub fn template_keys(&self) -> Box<dyn Iterator<Item = &TemplateKey> + '_> {
```

Comment:
re `TODO`: do you mean to get rid of `dyn`? If so, I think having `dyn` here is probably fine.

Reply:
What I meant is [this](https://github.com/0xPolygonMiden/miden-base/pull/1015#issuecomment-2573934619), basically

Reply:
Let's create an issue for this. it sounds like a good idea, but I'm not sure this will be possible in all circumstances.

Reply:
Created https://github.com/0xPolygonMiden/miden-base/issues/1051